src/sif/autocost.cc

```diff
@@ -473,9 +486,12 @@ Cost AutoCost::TransitionCost(const baldr::DirectedEdge* edge,
     if (!edge->has_flow_speed() || flow_mask_ == 0)
       seconds *= trans_density_factor_[node->density()];
 
-    c.cost += shortest_ ? 0.f : seconds;
+    c.cost += seconds;
   }
 
+  // Account for the user preferring distance
+  c.cost *= inv_distance_factor_;
```

Comment:
this is also a linear combination for the transition cost however we can simplify it because the distance of a transition is always 0 which simplifies away the second half of the formula